frontend/document/documentregistry.lua

```diff
@@ -21,6 +22,10 @@ function DocumentRegistry:addProvider(extension, mimetype, provider, weight)
         weight = weight or 100,
     })
     self.filetype_provider[extension] = true
+    -- We take the first extension registered for a mimetype as canonical.
+    -- Provided we order the calls to addProvider() correct, that means
+    -- epub instead of epub3, etc.
```

Comment:
It's too late. :-P
```suggestion
    -- We regard the first extension registered for a mimetype as canonical.
    -- Provided we order the calls to addProvider() correctly, that means
    -- epub instead of epub3, etc.
```